Build failure: pkgsCross.xxxx.stow

### Steps To Reproduce

Steps to reproduce the behavior:
1. Build `pkgsCross.xxxx.stow`. E.g.:
    ``` sh
    nix-build -I nixpkgs=channel:nixpkgs-unstable '<nixpkgs>' -A pkgsCross.aarch64-multiplatform.stow
    ```

### Build log

```
checking for perl... no
checking for perl5... no
configure: error: Perl not found; check your $PATH.
```

### Additional context

The problem occurs because adding perl to `buildInputs` does not add it to `nativeBuildInputs` for cross compilation. I'm not entirely sure if perl is needed to build the derivation, but adding perl to nativeBuildInputs solves the problem:
```nix
nativeBuildInputs = [ perl ];
buildInputs = with perlPackages; [ perl IOStringy TestOutput ];
```
Note that this issue might exist for other packages too.

Related forum post: https://discourse.nixos.org/t/cross-compiling-perl-packages-like-stow-riscv-aarch/39221

### Notify maintainers

@sarcasticadmin

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-47-generic, Ubuntu, 22.04.4 LTS (Jammy Jellyfish), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/misc/stow/default.nix
+++ b/pkgs/tools/misc/stow/default.nix
@@ -14,6 +14,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-p7rGc9qiRs9U/7SKET6t0gSmFWMOdrU7X2TYGidbd7M=";
   };
 
+  nativeBuildInputs = [ perlPackages.perl ];
   buildInputs = with perlPackages; [ perl IOStringy TestOutput ];
 
   doCheck = true;

--- a/pkgs/tools/misc/stow/default.nix
+++ b/pkgs/tools/misc/stow/default.nix
@@ -1,7 +1,6 @@
 { lib
 , stdenv
 , fetchurl
-, perl
 , perlPackages
 }:
 

